### PR TITLE
fix: checkbox styles for display only mode

### PR DESCRIPTION
**Pull request by DeepakSap14**
## Related Issue
Closes SAP/fundamental-styles#
https://github.com/SAP/fundamental-styles/issues/837

## Description
This is Styles changes for Display Only checkbox.

### After:
Display only checkbox

### CODE REVIEW

**Comment by JKMarkowski** [comment]
Hello @DeepakSap14, could you add example of usage to documentation page `checkbox.md`?

File: src/checkbox.scss [hunk 1]
```diff
@@ -169,3 +170,61 @@ $block: #{$fd-namespace}-checkbox;
     }
   }
 }
+
+.#{$displayonlyblock} {
```
Comment (JKMarkowski) [inline src/checkbox.scss:174]: I would go more for `.is-readonly / readonly` state instad of having another block class. So class: `fd-checkbox is-readonly`. And on css `&.is-readonly + fd-checkbox__label {}`

File: src/checkbox.scss [hunk 2]
```diff
@@ -169,3 +170,61 @@ $block: #{$fd-namespace}-checkbox;
...
+
+  $fd-displaycheckbox-margin: 0.5rem;
+
+  &__label {
+    width: fit-content;
+    min-width: fit-content;
+    overflow: hidden;
+    display: flex;
+    pointer-events: none;
```
Comment (JKMarkowski) [inline src/checkbox.scss:183]: fit-content doesn't work on ie11, Width doesn't need to be changed

File: src/checkbox.scss [hunk 3]
```diff
@@ -6,6 +6,7 @@
 .fd-checkbox
 */
 $block: #{$fd-namespace}-checkbox;
+$displayonlyblock: #{$fd-namespace}-displayonly-checkbox;
```
Comment (InnaAtanasova) [inline src/checkbox.scss:9]: I don't think this should be a component. Display-only mode should be a modifier class. Example:
`fd-checkbox--display-only`

**Comment by jbadan** [comment]
This should be marked `fix` if you've made no changes to the existing underlying html - see https://github.com/SAP/fundamental-styles/wiki/Automation-Processes---Versioning-and-Publishing#zeroth-version-library

**Comment by DeepakSap14** [comment]
Hi,
i have updated checkbox.md file as well. but while creating example in checkbox , issue with indeterminate checkbox is found and styles not working as expected.(previous implementation as well) please check once.
Screenshot

File: docs/pages/components/checkbox.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+{% include display-component.html component=checkbox-rtl %}
+
+## Display Only Checkbox
+
+{% capture checkbox-rtl %}
+    <input type="checkbox" class="fd-checkbox--display-only" id="1tn4O446" checked style="display:none">
+	<label class="fd-checkbox--display-only__label" for="1tn4O446">Label</label>
+  
+    <input type="checkbox" class="fd-checkbox--display-only" id="StRHi735"  style="display:none">
+	<label class="fd-checkbox--display-only__label" for="StRHi735">Label</label>
```
Comment (jbadan) [inline docs/pages/components/checkbox.md:176]: I think this is a little backwards, I would think it should be something like `fd-checkbox__label fd-checkbox__label--display-only`.

File: docs/pages/components/checkbox.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
+---
```
Comment (jbadan) [inline docs/pages/components/checkbox.md:1]: If we're going to move checkboxes to it's own page (👍 ), we should remove the examples from the "Form" page and provide a link to this page in it's place.

File: docs/pages/components/checkbox.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        <label class="fd-checkbox__label" for="h4z5N521">Label</label>
+    </div>
+{% endcapture %}
+
+{% include display-component.html component=checkbox-rtl %}
+
+## Display Only Checkbox
+
+{% capture checkbox-rtl %}
+    <input type="checkbox" class="fd-checkbox--display-only" id="1tn4O446" checked style="display:none">
```
Comment (jbadan) [inline docs/pages/components/checkbox.md:172]: This should be `fd-checkbox fd-checkbox--display-only`. It should always have `fd-checkbox`

File: docs/pages/components/checkbox.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+{% include display-component.html component=checkbox %}
+
+## Display Only Checkbox
+{% capture display-checkbox %}
+<fieldset class="fd-fieldset">
+    <legend class="fd-fieldset__legend">Display Only Checkboxes</legend>
+    <div class="fd-form-group">
+        <div class="fd-form-item">
+            <input type="checkbox" class="fd-checkbox fd-checkbox--display-only" id="1tn4O446" checked>
+            <label class="fd-checkbox__label--display-only" for="1tn4O446">Label</label>
```
Comment (jbadan) [inline docs/pages/components/checkbox.md:278]: These should also have class `fd-checkbox__label`:  `fd-checkbox__label fd-checkbox__label--display-only`

**Comment by DeepakSap14** [comment]
@jbadan @InnaAtanasova 
In fiori for Display only mode checkbox, as font-family, HTML Unicode etc values are defined. so i picked it to develop as Css. (same way for other checkboxes also these values are defined).
If we go by icon approach, we have to make major changes in fundamental-ngx/core checkbox implementation to accommodate Display Only checkbox icon.

**Comment by jbadan** [comment]
What is the purpose of display only checkboxes? The spec isn't very clear to me. My assumption is that because read-only checkboxes are not considered an accessible pattern, that they've introduced this concept of display only. If that is the case, what is your plan to make implementing this accessible + not allowing interaction with the checkbox?

**Comment by InnaAtanasova** [comment]
Closing this PR for now as we need further discussions with designers and UX team.

**Pull request closed without merging**